crates/starknet-devnet-server/src/api/json_rpc/mod.rs

```diff
@@ -1063,6 +1064,62 @@ mod requests_tests {
         }
     }
 
+    #[test]
+    fn deserialize_devnet_methods_with_optional_body() {
+        for mut body in [
+            json!({
+                "method": "devnet_dump",
+                "params": {}
+            }),
+            json!({
+                "method":"devnet_dump",
+            }),
+            json!({
+                "method":"devnet_dump",
+                "params": {"path": "path"}
+            }),
+            json!({
+                "method":"devnet_getPredeployedAccounts",
+                "params": {"with_balance": true}
+            }),
+            json!({
+                "method":"devnet_getPredeployedAccounts",
+            }),
+            json!({
+                "method":"devnet_getPredeployedAccounts",
+                "params": {}
+            }),
+            json!({
+                "method":"devnet_postmanFlush",
+                "params": {"dry_run": true}
+            }),
+            json!({
+                "method":"devnet_postmanFlush",
+            }),
+            json!({
+                "method":"devnet_postmanFlush",
+                "params": {}
+            }),
+        ] {
+            let mut json_rpc_object = json!({
+                "jsonrpc": "2.0",
+                "id": 1,
+            });
+
+            json_rpc_object.as_object_mut().unwrap().append(body.as_object_mut().unwrap());
+
+            let RpcMethodCall { method, params, .. } =
+                serde_json::from_value(json_rpc_object).unwrap();
+            let params: serde_json::Value = params.into();
+            let deserializable_call = serde_json::json!({
+                "method": &method,
+                "params": params
+            });
+
+            assert_deserialization_succeeds(deserializable_call.to_string().as_str())
```

Comment:
Why do we have to make four conversions in order to test?

Reply:
because this is the way the JSON-RPC request is being deserialized into `JsonRpcRequest` enum. Copied logic from rpc_handler. Values which are empty or non existent, get deserialzied to serde_json::Value::Null

Reply:
It reminded me a lot of the production logic, just was hoping that the tests could be simpler. But if not, I'll resolve it.

Reply:
they can be, but the flow will not simulate how the request is deserialised. This is important in executing the test successfully